src/main/java/nodebox/network/webSocketClientEndpoint.java

```diff
@@ -0,0 +1,120 @@
+package nodebox.network;
+
+import java.net.URI;
+import java.nio.ByteBuffer;
+import javax.websocket.ClientEndpoint;
+import javax.websocket.CloseReason;
+import javax.websocket.ContainerProvider;
+import javax.websocket.OnClose;
+import javax.websocket.OnMessage;
+import javax.websocket.OnOpen;
+import javax.websocket.Session;
+import javax.websocket.WebSocketContainer;
+import javax.websocket.DeploymentException;
+
+import java.io.IOException;
+import java.util.concurrent.*;
+//import java.util.UUID;
+//import java.util.concurrent.ConcurrentLinkedQueue;
+//import java.util.concurrent.ConcurrentHashMap;
+//import static java.lang.System.out;
+/*
+Flow:
+Whenever a node wants to send websocket data it adds a value to the OUT map by calling sendMessage(), send message returns a GUID which is also the key for the map entry
+The node then blocks in a loop quierying the IN map with the GUID given to it
+Meanwhile another thread is constantly looking at the OUT map and sends websocket data when a new map entry is created
+the same thread is also reading websocket data and adding to the IN queue.
+the same thread also picks up on special incoming request for application control
+
+ */
+
+@ClientEndpoint
+public class webSocketClientEndpoint {
+    public boolean isConnected = false;
+    Session userSession = null;
+    private MessageHandler messageHandler;
+    private URI endpointURI;
+    private final ExecutorService pool = Executors.newFixedThreadPool(10);
+
+    private Future<Session> asyncConnectToServer(Object annotatedEndpointInstance, URI uri) {
+        return pool.submit(new Callable<Session>() {
+            @Override
+            public Session call() throws Exception {
+                try {
+                    WebSocketContainer container = ContainerProvider.getWebSocketContainer();
+                    return(container.connectToServer(annotatedEndpointInstance, uri));
```

Comment:
This line causes the code to fail compiling. This is the error I get:

    webSocketClientEndpoint.java:45: error: local variable annotatedEndpointInstance is accessed from within inner class; needs to be declared final
        [javac]                     return(container.connectToServer(annotatedEndpointInstance, uri));